core/src/peer_id.rs

```diff
@@ -164,6 +167,60 @@ impl From<PeerId> for Vec<u8> {
     }
 }
 
+#[cfg(feature = "serde")]
+impl Serialize for PeerId {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: _serde::Serializer,
+    {
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&self.to_base58())
+        } else {
+            serializer.serialize_bytes(&self.to_bytes()[..])
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for PeerId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: _serde::Deserializer<'de>,
```

Comment:
```suggestion
        D: serde::Deserializer<'de>,
```

Nit pick, would this work as well?